src/hotspot/share/gc/g1/g1ParScanThreadState.hpp

```diff
@@ -83,6 +84,8 @@ class G1ParScanThreadState : public CHeapObj<mtGC> {
   // Size (in elements) of a partial objArray task chunk.
   int _partial_objarray_chunk_size;
   PartialArrayTaskStepper _partial_array_stepper;
+  // Used to check whether string dedup should be applied to an object.
+  Klass* _string_klass_or_null;
```

Comment:
Maybe add a note that we replicate java_lang_String::is_instance_inlined() here for performance reasons (skipping the NULL check), so that when it changes, maybe a grep will find this here?

Reply:
I think such a comment is better placed at the point of use, which is what I intend to do.